src/useEvent.ts

```diff
@@ -15,21 +16,17 @@ const useBrowserLayoutEffect = typeof window !== "undefined" ? useLayoutEffect :
  * - Properties or state accessed within the callback will always be "current"
  */
 export function useEvent<TCallback extends AnyFunction>(callback: TCallback): TCallback {
-  // Keep track of the latest callback:
-  const latestRef = useRef<TCallback>(useEvent_shouldNotBeInvokedBeforeMount as any);
-  useBrowserLayoutEffect(() => {
-    latestRef.current = callback;
+  const ref = useRef(useEvent_shouldNotBeInvokedBeforeMount as TCallback);
+  useBrowserEffect(() => {
+    ref.current = callback;
   }, [callback]);
 
-  // Create a stable callback that always calls the latest callback:
-  const stableRef = useRef<TCallback>(null as any);
-  if (!stableRef.current) {
-    stableRef.current = function(this: any) {
-      return latestRef.current.apply(this, arguments as any);
-    } as TCallback;
-  }
-
-  return stableRef.current;
+  return useCallback<TCallback>(
```

Comment:
I prefer using the `useRef` approach, because it avoids creating a new function instance and empty dependency array every render. Tiny optimization, but worthwhile for such a low level hook.

Reply:
Ah makes sense i did not understand the second useRef before. Will change it back to this approach.